Checklist done

PedroTrujilloV opened the pull request:
I can’t test the confirmation view controller because is showing for me the address error. I created a pdf version of the image, for the alert controller.

PedroTrujilloV committed:
Merge remote-tracking branch 'cforlando/master'

PedroTrujilloV committed:
test avoid error

error: Your local changes to the following files would be overwritten
by merge:
	Street Trees.xcodeproj/project.pbxproj
Please, commit your changes or stash them before you can merge.
Aborting
 (128)

PedroTrujilloV committed:
Merge remote-tracking branch 'cforlando/master'

PedroTrujilloV committed:
Merge remote-tracking branch 'cforlando/master'

PedroTrujilloV committed:
Confirmation Page view controller and story board for this.

PedroTrujilloV committed:
STConfirmationPageViewController:
- Table view methods
- Get information from Wufoo form

PedroTrujilloV committed:
STConfirmationPageViewController

Story board View Controller and Class

PedroTrujilloV committed:
Cell registration

cell registration for each field in the form

PedroTrujilloV committed:
fixed the mistake

a miss understand about the Wufoo form and the three objects.

PedroTrujilloV committed:
Protocol ready

- Confirmation Page
- Protocol in confirmation page with function at Order Form Navigation
controller setting the var wufooForm

PedroTrujilloV committed:
Done (no tested)

There requirements are done, but running the app shows :2016-10-19
00:36:33.508 Street Trees[41864:1657473] Can't find keyplane that
supports type 5 for keyboard iPhone-Portrait-PhonePad; using
3561727473_Portrait_iPhone-Complex-Pad_Default
fatal error: An Address has not been passed.: file
/Users/Pedro/energy-saving-trees-ios/Street
Trees/STUserDetailsFormViewController.swift, line 160

PedroTrujilloV committed:
Done:
- func confirmationFormViewController(form: STConfirmationPageViewController, didCompleteWithWufooForm aWufooForm: STTKWufooForm
- Remove test values
- remove this function as the table view will should use a datasource to insert its content.
- remove test data configuration
- remove unused function
- remove commented code
- Create an enum for the cells and then create a datasource property or function that returns a data source struct that contains your content for the tableview. You should be able to query your new struct for the amount of sections and query for how many items are in a section.
- Use the datasource struct return the correct amount of rows for the section.
- create a single cell in the tableview on the storyboard. looking at your code, a basic tableviewcell would be fine. Then you can query your datasource struct for the correct cell information.

I would ask the datasource for the enum at section and row and then create a switch for each enum variable instead of creating a ton of if statements. a large list of if statements can be rather cpu intensive and we don't want the cpu to do any additional work than what it has too.

-  only assign the delegate in this if let. By conforming the confirmationViewController to the TreeDescription, Address, and Contact protocols it will be given the correct objects when it is placed on the navigation stack.
- Revert this to use swift 2.2. We do not support Xcode 8 or Swift 2.3 at this time.
- This review label is not required as the text in the navigation bar will be fine.

PedroTrujilloV committed:
Done
-remove this button and add a bar item to the right hand side of the navigation bar to keep it inline with the rest of the form. This is a deviation from the designs.

PedroTrujilloV committed:
add image pdf to alert controller

PedroTrujilloV committed:
PR Commit

PedroTrujilloV committed:
change image

added the image “sent.jpg” from Tom.

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+            textLabel = treeDescription?.name ?? "Error retrieving tree name"
+            imageView = treeDescription?.image() ?? UIImage(named: "blankMap")
+        case .Address:
+            textLabel = (address?.streetAddress)!  + " " +  (address?.secondaryAddress)! ?? "Error retrieving address"
```
Never use !. It can cause all kinds of problems.

LondonAtlas replied:
Forcing unwraps will causes crashes and make the ?? useless.

Properties don't require () around them when you are string to concatenate them into a string.
address?.streetAddress!  + " " +  address?.secondaryAddress!
or
"(address?.streetAddress!)  (address?.secondaryAddress!)"

Better yet:
let address = address?.streetAddress ?? "No address specified"
let address2 = address?.secondaryAddress ?? "" // 2nd address not required so convert nil to an empty string.
textLabel = "(address) (address2)"

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+    @IBAction func confirmButtonItemTouchUpInside(sender: AnyObject)
+    {
+        
+        if confrim()
```
always use self when calling a function in the current class.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+    //******************************************************************************************************************
+    // MARK: - Private Functions
+    
+    func confrim() -> Bool
```
type-o

"confirm"

LondonAtlas replied:
Also what are we confirming?

PedroTrujilloV replied:
To guard a the variables is better than force an unwrap, like you told me.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+        return self.dataSource[anIndex].count
+    }
+    
+    func item(forRow aRow: Int, inSection aSection: Int) -> STConfirmationDetailRows {
```
Pass an NSIndexPath

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+    case Phone
+    case email
+    
+    func sectionHeader() -> String?
```
String doesn't need to be an optional since all cases are accounted for.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+    case CityStateZip
+    case UserName
+    case Phone
+    case email
```
Inconsistent naming.

"Email"

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+//
+
+import UIKit
+import StreetTreesTransportKit
```
Alpha order the imports

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+{
+    
+    var treeDescription:STPKTreeDescription?
+    var address: STTKStreetAddress?
```
alpha order variables

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+    
+    weak var  delegate:STConfirmationPageViewControllerDelegate?
+
+    // MARK: - Table View Methods:
```
Use the sectioning comments that can be found in the other files.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+        return dataSource.sections
+    }
+    
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int
```
Alpha order functions

LondonAtlas reviewed (changes requested):
Looking good. Some changes required.

LondonAtlas commented:
App crashes immediately when trying to move to the Confirm view controller.

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+    
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int
+    {
+        return dataSource.sections
```
Always use self when referencing class properties.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+        }
+        
+        cell.textLabel?.text = textLabel
+        cell.imageView?.image = imageView
```
imageView is only ever added once. Move this cell.imageView?.image to the .TreeName case.

PedroTrujilloV replied:
Done

LondonAtlas commented:
UI is not correct
<img width="487" alt="screen shot 2016-10-30 at 4 25 17 pm" src="https://cloud.githubusercontent.com/assets/1647012/19839729/8f291b06-9ebd-11e6-8e4c-c4c8a62efc41.png">

Please match this:
<img width="333" alt="screen shot 2016-10-30 at 4 26 47 pm" src="https://cloud.githubusercontent.com/assets/1647012/19839742/c54be826-9ebd-11e6-9049-b6be46a662b7.png">

PedroTrujilloV commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,179 @@
+    
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int
+    {
+        return dataSource.sections
```
Done

PedroTrujilloV committed:
Some changes required

Done:

- Alpha order the imports
- Inconsistent naming.
- String doesn't need to be an optional since all cases are accounted
for.
- Pass an NSIndexPath
- alpha order variables
- Use the sectioning comments that can be found in the other files.
- Never use !. It can cause all kinds of problems.
- always use self when calling a function in the current class.
- type-o “confirm”
- Always use self when referencing class properties.
- imageView is only ever added once. Move this cell.imageView?.image to
the .TreeName case.
working in:

- Alpha order functions
- correct UI

PedroTrujilloV committed:
Merge remote-tracking branch 'cforlando/master'

# Conflicts:
#	Street Trees/OrderForm.storyboard

PedroTrujilloV committed:
fixed more forced unwraps

PedroTrujilloV committed:
some fixing

PedroTrujilloV committed:
UI: add font properties to section

PedroTrujilloV committed:
Merge branch 'cforlando/master'

# Conflicts:
#	Street Trees/OrderForm.storyboard

LondonAtlas commented on Street Trees/OrderForm.storyboard:
```diff
@@ -60,14 +60,44 @@
                                 <color key="textColor" red="0.0" green="0.0" blue="0.0" alpha="1" colorSpace="calibratedRGB"/>
                                 <nil key="highlightedColor"/>
                             </label>
+<<<<<<< HEAD
```
Conflict city.

akozlik replied:
Great catch.

PedroTrujilloV replied:
Done

PedroTrujilloV committed:
delete order form

PedroTrujilloV committed:
added OrderForm reference again

PedroTrujilloV committed:
UI:
- Footer space  between sections.
- Font: size, color line space, for header.

Problems retrieving information to show.

PedroTrujilloV committed:
fixed UI:

- Loading information from nav controller. (moved Confirm..VC to Order...storyb...)
- header gets stuck, not any more. (changed table property from Plain to Grouped )

PedroTrujilloV committed:
UI:

- Done

Deleted Confirmation... storyboard

PedroTrujilloV commented:
UI Done.
